Removed redundant subnetmask in peers.j2 template

tungsec opened the pull request:
# Description

<!---
Please include a summary of the change and which issue is fixed.
--->

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Small minor change not affecting the Ansible Collection/Role code (Github Actions Workflow, Documentation etc.)

## How Has This Been Tested?
<!---
Please describe the tests that you ran to verify your changes.
Create a PR into `main` branch.
--->

TVerkuijlenCG committed:
Removed redundant subnetmask in peers.j2 template

MonolithProjects commented on roles/wireguard/templates/peers.j2:
```diff
@@ -5,7 +5,7 @@
 - WireGuardPeer:
   - FriendlyName: {{ peer.name }}
-  - AllowedIPs: {{ peer.allowed_ip }}/32
+  - AllowedIPs: {{ peer.allowed_ip }}
```
In my usecase i am setting always only single IP address as `AllowedIPs`. That's why is the `/32` netmask hardcoded in the template. As a result, i don't need write netmask for every IP in peers configuration in the group vars.

But can see the point that if you want to set a different netmask, you don't have this option here. For backward compatibility could you please change the line to something like this?
```
- AllowedIPs: {{ peer.allowed_ip }}{% if not '/' in peer.allowed_ip %}/32{% endif %}
```

tungsec replied:
Ok I get your point. I have updated it with your suggestion.

tungsec replied:
The tests fail now because the if-statement removes the newline from the AllowIPs line (raw):
"---\n#\n# This is simple jinja2 file used to generate peer variable based on dynamic input.\n#\n  - WireGuardPeer:\n    - FriendlyName: user1\n    - AllowedIPs: 10.213.213.2/32    - PublicKey: \"xxxx=\"\n    - PresharedKey: \"xxxx=\n\"\n    - PersistentKeepalive: 25\n"

tungsec replied:
There is an option to add "#jinja2: trim_blocks:False" at the top of the template. Ref: https://stackoverflow.com/questions/22350175/how-do-i-get-an-ansible-template-to-honor-new-lines-after-a-conditional

MonolithProjects replied:
Seems like there is no option how to set `trim_block` directly for `lookup` plugin. So, i think we can go with the first option. Please add `#jinja2: trim_blocks:False` to the template file. Thanks :)

Commit:
Update peers.j2

As request by MonolithProjects make AllowedIPs dynamic based on allowed_ip subnet mask.

Commit:
Set trim_block:False in template file to prevent newline being removed

The molecule tests failed because the if-statement I added removes the newline from the AllowIPs line (raw):
"---\n#\n# This is simple jinja2 file used to generate peer variable based on dynamic input.\n#\n - WireGuardPeer:\n - FriendlyName: user1\n - AllowedIPs: 10.213.213.2/32 - PublicKey: "xxxx="\n - PresharedKey: "xxxx=\n"\n - PersistentKeepalive: 25\n".
Adding the "#jinja2: trim_blocks:False" prevents the removal of the newline.
